### PR TITLE
[Small Feature] Reworked CGAL::Color

### PR DESCRIPTION
**Pull request by sgiraudot**
## Rationale

The `CGAL::Color` class is badly documented and is not compatible with header-only mode (see https://github.com/CGAL/cgal/issues/3620).

This small feature makes it compliant to header-only, merges the fake `.h` of the doc with the actual header, makes the doc consistent and adds a few useful functions.

## Summary of API changes

- The `alpha` extra-parameter is now documented and defaults to 255
- Additional methods that return references to the data members are provided
- For convenience, `operator[](std::size_t)` is provided
- Additional methods `to_rgba()` and `to_rgb()` return the underlying raw container (array of size 4 or 3)
- Additional method `to_hsv()` computes the HSV conversion of the RGB color and returns the values in an array
- Additional method `set_hsv()` does the inverse conversion to set RGB color from HSV values
- Additional method `set_rgb()` allows to change directly all members
- Global variables `CGAL::BLACK`, `CGAL::BLUE`, etc. are replace by inline functions `CGAL::black()`, `CGAL::blue()`, etc.
- The typedefs `CGAL::Classification::RGB_Color` and `CGAL::Classification::HSV_Color` are removed: `CGAL::Color` is now used along with the HSV/RGB conversion operators.

(For convenience, this PR also removes the deprecated parts of Classification. These parts have been deprecated in 4.13 and 4.14 and can thus be removed in 4.15.)

## License and copyright ownership

(no change)

## CHANGES.md

TODO

## Status

-   Developed and locally tested (GNU/Linux)
-   (Empty) small feature page: [Reworked CGAL::Color](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Reworked_CGAL::Color)
- Pre-approval date:  2018/02/25

## Submission

-  [`CGAL::Color`](https://cgal.geometryfactory.com/~sgiraudot/CGAL_Color/Stream_support/classCGAL_1_1Color.html)
-  [`CGAL::Classification::Feature::Color_channel`](https://cgal.geometryfactory.com/~sgiraudot/CGAL_Color/Classification/classCGAL_1_1Classification_1_1Feature_1_1Color__channel.html) (the only change in Classification due to this PR)

### CODE REVIEW

**Comment by afabri** [comment]
@sgiraudot please build the html documentation and put it online so that it can be reviewed.

**Comment by sgiraudot** [comment]
It's updated.

File: Stream_support/include/CGAL/IO/Color.h [hunk 1]
```diff
@@ -23,83 +23,326 @@
...
+  unsigned char a() const { return alpha(); }
+  /// \endcond
+  
+  /// @}
+
+  /// \name Array Access
+  /// @{
+
+  /*!
+    returns the i^{th} component of the rgb color (the 0^{th} is red,
```
Comment (sloriot) [inline Stream_support/include/CGAL/IO/Color.h]: i-th?

**Comment by afabri** [comment]
@maxGimeno can you add this to the testsuite, as it was pre-approved end of February.

File: Stream_support/include/CGAL/IO/Color.h [hunk 2]
```diff
@@ -23,83 +23,326 @@
...
+
+  /*!
+    returns the array with rgb values.
+  */ 
+  const cpp11::array<unsigned char, 3>& to_rgb() const
   {
-    return ( (red() == c.red()) &&
-             (green() == c.green()) &&
-             (blue() == c.blue()) );
+    return reinterpret_cast<const cpp11::array<unsigned char, 3>&>(m_data);
```
[the remark this answers is not in the document]
Comment (afabri) [reply Stream_support/include/CGAL/IO/Color.h]: @sgiraudot It is approved, but you write "probably one is enough", so do you still change it?

**Comment by afabri** [comment]
@lrineau Do you agree that it is enough to write "Color  - breaking changes after an overhaul."  in the CHANGES.md file.

**Comment by maxGimeno** [comment]
Done. Is the TODO done ?

**Comment by afabri** [comment]
> 
> 
> Done. Is the TODO done ?

Yes, I added an entry to CHANGES.md

**Comment by maxGimeno** [comment]
There is a conflict for the file Classification/include/CGAL/Classification/Color.h with another branch, I remove this from integration for now.

**Comment by afabri** [comment]
@maxGimeno with which branch?

**Comment by maxGimeno** [comment]
The branch about cpp14 in #3781

**Comment by lrineau** [comment]
There is now an official conflict with `master`.

**Comment by maxGimeno** [comment]
I am quite confident this PR is the reason of [the errors in Installation](https://cgal.geometryfactory.com/CGAL/testsuite/results-5.0-Ic-9.shtml#Installation).
The test that fails is the following:
```cpp
  volatile const CGAL::Color* c = &CGAL::BLACK;
  
  return (c != 0) ? 0 : 1;
```

**Comment by sgiraudot** [comment]
This test is supposed to be changed by this PR (cf https://github.com/CGAL/cgal/pull/3625/files#diff-d1b3d31c91876cd2e356040d5d1c1f4d). I've retested it locally (also in not-header only mode) and it's working, I can't reproduce the bug. Is it possible that something wrong happened when this was pushed to integtration?

**Comment by sgiraudot** [comment]
My bad, it's fixed now (thanks to @maxGimeno for noticing…).

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.0-Ic-10.shtml

**Pull request merged**